### PR TITLE
Add remaining parameter types to docker.api.container logs

### PR DESCRIPTION
**Pull request by adamtheturtle**

### CODE REVIEW

File: stubs/docker/docker/api/container.pyi [hunk 1]
```diff
@@ -68,15 +69,15 @@ class ContainerApiMixin:
     def kill(self, container, signal: Incomplete | None = None) -> None: ...
     def logs(
         self,
-        container,
+        container: str,
```
Comment (adamtheturtle) [inline stubs/docker/docker/api/container.pyi:72]: This is documented at https://github.com/docker/docker-py/blob/b6464dbed92b14b2c61d5ee49805fce041a3e083/docker/api/container.py#L831 as:

```
            container (str): The container to get logs from
```

However, I'm not sure that this is right - in other methods in this class, `container` is defined as:

```
container (str or dict)
```

And those methods seem to use `container` in the same way.

In `wait` in particular, the doctoring says:

```
container (str or dict): The container to wait on. If a dict, the
                ``Id`` key is used.
```

Comment (srittau) [reply stubs/docker/docker/api/container.pyi:72]: See https://github.com/docker/docker-py/blob/b6464dbed92b14b2c61d5ee49805fce041a3e083/docker/api/container.py#L894: A dict wouldn't work here.

Comment (adamtheturtle) [reply stubs/docker/docker/api/container.pyi:72]: That line is:

```python
        url = self._url("/containers/{0}/logs", container)
```

I don't see how that is different to the line in `wait` https://github.com/docker/docker-py/blob/b6464dbed92b14b2c61d5ee49805fce041a3e083/docker/api/container.py#L1338:

```python
        url = self._url("/containers/{0}/wait", container)
```

Comment (srittau) [reply stubs/docker/docker/api/container.pyi:76]: Good point. It seems the `check_resource` decorator unwraps a dict, so this should work here as well.

Comment (adamtheturtle) [reply stubs/docker/docker/api/container.pyi:76]: Do you think it is better to match the implementation (accept a `dict` or `str`) or the documentation (just `str`)?

Comment (srittau) [reply stubs/docker/docker/api/container.pyi:76]: We usually match the implementation in typeshed, especially in cases like this, where the missing dict seems to be an oversight in the docstring. (Upstream might be happy about a PR.)

Comment (adamtheturtle) [reply stubs/docker/docker/api/container.pyi:76]: And - if the former - do you have any tips on whether / how I can say "the dict must have an `Id` field"?

Comment (adamtheturtle) [reply stubs/docker/docker/api/container.pyi:76]: So far upstream has not been responsive to any of my related PRs, so I won't make a PR there.

Comment (srittau) [reply stubs/docker/docker/api/container.pyi]: We can use a `TypedDict` for that purpose:

```python
@type_check_only
class _HasId(TypedDict):
    Id: str
```

**Review by srittau — APPROVED** [review]

File: stubs/docker/docker/api/container.pyi (hunk 1 above)
Comment (JelleZijlstra) [reply stubs/docker/docker/api/container.pyi]: Note though that a `dict[str, Any]` wouldn't match this TypedDict.

Comment (adamtheturtle) [reply stubs/docker/docker/api/container.pyi]: @JelleZijlstra Is there a change you'd like me to make in a follow up?

**Pull request merged**